pkg/storage/utils/decomposedfs/node/locks.go

```diff
@@ -121,7 +121,17 @@ func (n Node) ReadLock(ctx context.Context) (*provider.Lock, error) {
 		appctx.GetLogger(ctx).Error().Err(err).Msg("Decomposedfs: could not decode lock file, ignoring")
 		return nil, errors.Wrap(err, "Decomposedfs: could not read lock file")
 	}
-	return lock, err
+
+	// lock already expired
+	if lock.Expiration != nil && time.Now().After(time.Unix(int64(lock.Expiration.Seconds), int64(lock.Expiration.Nanos))) {
+		if err = os.Remove(f.Name()); err != nil {
+			return nil, errors.Wrap(err, "Decomposedfs: could not remove expired lock file")
+		}
+		// we successfully deleted the expired lock
+		return nil, nil
```

Comment:
Is this correct? When I call `ReadLock` I would expect to get an `error` or a working lock.

Reply:
Confusing... This method only reads an existing Application-Lock file and returns the content. If the lock is expired, it is indeed correct to return nothing.

What you mean is `RefreshLock` or `SetLock`.

The method documentation should be more verbose on that.

Reply:
Mmh. I still don't get it. What would it return if the requested lock does not exist?

Reply:
An expired lock doesn't even exist from an outside perspective. We just don't have a collector for expired locks. Therefore we delete it lazily when someone calls `ReadLock`

Reply:
Yes I got that. But in that case it should return `not found` imho, because `An expired lock doesn't even exist from an outside perspective.`